src/server/plugins/Plugin.js

```diff
@@ -18,7 +18,9 @@ module.exports = class Plugin {
     this.uiExportsSpecs = opts.uiExports || {};
     this.requiredIds = opts.require || [];
     this.version = opts.version || pkg.version;
-    this.publicDir = opts.publicDir !== false ? resolve(path, 'public') : null;
+    this.publicDir = opts.publicDir !== false
+        ? resolve(path, opts.publicDir || 'public')
+        : null;
     this.externalCondition = opts.initCondition || _.constant(true);
```

Comment:
The assignment logic here reads more complex than it is. Can you write this exactly how the description in the pull reads?

```
if (opts.publicDir == null) {
 this.publicDir = resolve(path, 'public');
} else if (opts.publicDir === false) {
 this.publicDir = null
} else {
 this.publicDir = resolve(path, opts.publicDir);
}
```